File: Array_Concatenation/concat.go
package arrayconcatenation

import "strconv"

func findTheArrayConcVal(nums []int) int64 {
	result := 0
	for len(nums) > 0 {
		i, j := 0, len(nums)-1

		if i != j {
			concat := strconv.Itoa(nums[i]) + strconv.Itoa(nums[j])
			concatVal, _ := strconv.Atoi(concat)
			result += concatVal
		} else {
			result += nums[i]
		}

		if len(nums) > 1 {
			nums = nums[1 : len(nums)-1]
		} else {
			break
		}
	}

	return int64(result)
}

File: Array_Concatenation/concat.cs
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public long FindTheArrayConcVal(int[] nums)
    {
        long result = 0;
        while (nums.Length > 0)
        {
            int i = 0,
                j = nums.Length - 1;

            if (i != j)
            {
                string concat = $"{nums[i]}{nums[j]}";
                int concatVal = int.Parse(concat);
                result += concatVal;
            }
            else
            {
                result += nums[i];
            }

            if (nums.Length > 1)
            {
                nums = nums.Skip(1).Take(nums.Length - 2).ToArray();
            }
            else
            {
                break;
            }
        }

        return result;
    }
}
